cmake/FindNLopt.cmake

```diff
@@ -0,0 +1,83 @@
+# Find NLopt library.
+# The following variables are set
+#
+# NLopt_FOUND
+# NLopt_INCLUDE_DIRS
+# NLopt_LIBRARIES
+#
+# It searches the environment variable $NLopt_PATH automatically.
+
+unset(NLopt_FOUND CACHE)
+unset(NLopt_INCLUDE_DIRS CACHE)
+unset(NLopt_LIBRARIES CACHE)
+unset(NLopt_LIBRARIES_RELEASE CACHE)
+unset(NLopt_LIBRARIES_DEBUG CACHE)
+
+if(CMAKE_BUILD_TYPE MATCHES "(Debug|DEBUG|debug)")
```

Comment:
why not use:
```cmake
string(TOLOWER CMAKE_BUILD_TYPE lowercase_CMAKE_BUILD_TYPE)
if (lowercase_CMAKE_BUILD_TYPE MATCHES "debug")
```

Reply:
I copied the `FindNLopt.cmake` script from libnest2d. Do you think it's better to leave a note that the script was taken from libnest2d, or should we adapt this script to our own code style?

Reply:
Good thing you caught that, I was just about to make these changes.

Reply:
Oops, replied to the wrong one.

Reply:
Discussed with rburema and we agreed that we can adjust the code style here.
Normally I wouldn't adjust the code style for things where there are no code style regulations because that destroys the commit history (i.e. something useful) for something that's really just an opinion-based thing. But since this repository is new, we can do that here.
We are not going to need to compare these scripts with their originals (since pynest2d is a brand new project) so there's no need to keep the scripts the same as their originals either.
Code style is changed here: 9689e9f616100952696b88bc5a294e9df9b429f7 . And that includes this improvement (in slightly different form), although that is not strictly just code style.